sql/src/test/java/io/crate/operation/projectors/InternalResultProviderFactoryTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to CRATE.IO GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.projectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.crate.core.collections.TreeMapBuilder;
+import io.crate.executor.transport.distributed.BroadcastDistributingDownstream;
+import io.crate.executor.transport.distributed.ModuloDistributingDownstream;
+import io.crate.executor.transport.distributed.SingleBucketBuilder;
+import io.crate.executor.transport.distributed.TransportDistributedResultAction;
+import io.crate.metadata.Routing;
+import io.crate.operation.NodeOperation;
+import io.crate.planner.node.dql.CollectPhase;
+import io.crate.planner.node.dql.MergePhase;
+import io.crate.planner.projection.Projection;
+import io.crate.planner.symbol.Symbol;
+import io.crate.test.integration.CrateUnitTest;
+import io.crate.types.DataType;
+import io.crate.types.LongType;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class InternalResultProviderFactoryTest extends CrateUnitTest {
+
+    private InternalResultProviderFactory resultProviderFactory;
+
+    @Before
+    public void before() {
+        DiscoveryNode discoveryNode = mock(DiscoveryNode.class);
+        when(discoveryNode.id()).thenReturn("local_node");
+        ClusterService clusterService = mock(ClusterService.class);
```

Comment:
btw there is a NoopClusterService which is pretty similar to what you achieve with the mocked clusterService here